packages/loot-core/src/server/main.ts

```diff
@@ -1139,6 +1160,60 @@ handlers['accounts-bank-sync'] = async function ({ id }) {
   return { errors, newTransactions, matchedTransactions, updatedAccounts };
 };
 
+handlers['simplefin-batch-sync'] = async function ({ ids }) {
+  const accounts = await db.runQuery(
+    `SELECT a.*, b.bank_id as bankId FROM accounts a
+         LEFT JOIN banks b ON a.bank = b.id
+         WHERE a.tombstone = 0 AND a.closed = 0 ${ids.length ? `AND a.id IN (${ids.map(() => '?').join(', ')})` : ''}
+         ORDER BY a.offbudget, a.sort_order`,
+    ids.length ? ids : [],
+    true,
+  );
+
+  let res;
+  try {
+    console.group('Bank Sync operation for all SimpleFin accounts');
+    res = await bankSync.SimpleFinBatchSync(
+      accounts.map(a => ({
+        id: a.id,
+        accountId: a.account_id,
+      })),
+    );
+  } catch (e) {
+    console.error(e);
```

Comment:
There is an issue here. If an exception occurs it logs it here, then just eats the exception. So then it goes on and tries to iterate `res` on 1187 and res is undefined.  So then you get another exception, which is caught by the caller.

Maybe that try/catch needs to be around the whole thing? What do we return here if an exception occurs?

Reply:
Fixed, this function shouldn't error out, it was a copy paste initial error from the old function